homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,204 @@
+"""EnaSolar solar inverter configuration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import pyenasolar
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CAPABILITY,
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_HOST,
+    CONF_MAX_OUTPUT,
+    CONF_NAME,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    DC_STRINGS,
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_SUN_DOWN,
+    DEFAULT_SUN_UP,
+    DOMAIN,
+    MAX_OUTPUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def enasolar_entries(hass: HomeAssistant):
+    """Return the hosts already configured."""
+    return {
+        entry.data[CONF_HOST] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class EnaSolarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User Configuration of EnaSolar Integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._enasolar = pyenasolar.EnaSolar()
```

Comment:
```suggestion
        self._enasolar: EnaSolar | None = None
```

Defer creating the object until the flow is started.  We want to avoid side effects in init if at all possible

Reply:
I am new to python so don't rally understand what is happening with this suggestion.

Reply:
OK, I think I now get this and will action.  __init__ in the EnaSolar class doesn't really do anything so moving the instantiation isn't hard.

Reply:
Hmm cant get tests to work if I move this,  as instantiation doesn't really do anything other than expose the methods, have gone back to plan A.